polyglot/piranha/src/models/piranha_arguments.rs

```diff
@@ -12,140 +12,175 @@ Copyright (c) 2022 Uber Technologies, Inc.
-  #[getset(get = "pub")]
+  #[get = "pub"]
+  #[builder(default)]
+  #[clap(skip)]
+  #[serde(default)]
   delete_consecutive_new_lines: bool,
   // User option that determines the prefix used for tag names that should be considered
   /// global i.e. if a global tag is found when rewriting a source code unit
   /// All source code units from this point will have access to this global tag.
-  #[getset(get = "pub")]
+  #[get = "pub"]
+  #[builder(default = "default_global_tag_prefix()")]
+  #[clap(skip)]
+  #[serde(default = "default_global_tag_prefix")]
   global_tag_prefix: String,
   /// Add a user option to configure the number of ancestors considered when applying
   /// parent scoped rules
-  #[getset(get = "pub")]
+  #[get = "pub"]
+  #[builder(default = "default_number_of_ancestors_in_parent_scope()")]
+  #[clap(skip)]
+  #[serde(default = "default_number_of_ancestors_in_parent_scope")]
   number_of_ancestors_in_parent_scope: u8,
   /// The number of lines to consider for cleaning up the comments
-  #[getset(get = "pub")]
+  #[get = "pub"]
+  #[builder(default = "default_cleanup_comments_buffer()")]
+  #[clap(skip)]
+  #[serde(default = "default_cleanup_comments_buffer")]
   cleanup_comments_buffer: usize,
   /// The AST Kinds for which comments should be deleted
-  #[getset(get = "pub")]
+  #[get = "pub"]
+  #[builder(default = "default_cleanup_comments()")]
+  #[clap(skip)]
+  #[serde(default = "default_cleanup_comments")]
   cleanup_comments: bool,
   /// Disables in-place rewriting of code
-  #[getset(get = "pub")]
+  #[get = "pub"]
+  #[builder(default = "default_dry_run()")]
+  #[clap(short = 'd', long, parse(try_from_str), default_value_t = false)]
+  #[serde(default = "default_dry_run")]
   dry_run: bool,
 }
 
-impl PiranhaArguments {
-  pub fn from_command_line() -> Self {
-    Self::new(CommandLineArguments::parse())
+impl From<PathBuf> for PiranhaArgumentsBuilder {
+
+  fn from(path_to_arguments_toml: PathBuf) -> Self {
+    let other: PiranhaArguments = read_toml(&path_to_arguments_toml, false);
+    let subs = other.substitutions();
+    let lang = get_language(other.get_language());
+    PiranhaArgumentsBuilder::default()
+      .path_to_code_base(other.path_to_code_base)
+      .path_to_configurations(other.path_to_configurations)
+      .path_to_output_summaries(other.path_to_output_summaries)
+      .input_substitutions(subs)
+      .dry_run(other.dry_run)
+      .language(other.language)
+      .delete_file_if_empty(other.delete_file_if_empty)
+      .delete_consecutive_new_lines(other.delete_consecutive_new_lines)
+      .global_tag_prefix(other.global_tag_prefix)
+      .cleanup_comments_buffer(other.cleanup_comments_buffer)
+      .cleanup_comments(other.cleanup_comments)
+      .number_of_ancestors_in_parent_scope(other.number_of_ancestors_in_parent_scope)
+      .piranha_language(lang)
+      .clone()
   }
+}
+
+
+
+impl From<CommandLineInput> for PiranhaArgumentsBuilder {
+
+  fn from(_:CommandLineInput) -> Self {
```

Comment:
`CommandLineInput` is a just a struct with no fields.
It represents "side effect of reading command line input". 
 I am creating this dummy `CommandLineInput` struct because `From` trait needs some type to parameterize upon. When we parse command line inputs it is a _side effect_ (because `PiranhaArguments::parse()` is magic :) ) so there is no way of showing it in type signature for `impl From<...>`.